src/libtls/tls_peer.c

```diff
@@ -286,6 +287,28 @@ static status_t process_server_hello(private_tls_peer_t *this,
 	}
 	extensions->destroy(extensions);
 
+	/* downgrade protection (see RFC 8446, section 4.1.3) */
+	memcpy(server_random_end, &this->server_random[24], 8);
+	if (this->tls->get_version_max(this->tls) == TLS_1_3 &&
+		version < TLS_1_3)
+	{
+		if (memcmp(server_random_end, tls_downgrade_protection_tls11, 8) == 0 ||
+			memcmp(server_random_end, tls_downgrade_protection_tls12, 8) == 0)
```

Comment:
There is a actually a `memeq()` helper to avoid the `== 0`, but with chunks it's even easier as you don't have to add the length again.
```suggestion
		if (chunk_equals(server_random_end, tls_downgrade_protection_tls11) ||
			chunk_equals(server_random_end, tls_downgrade_protection_tls12))
```